pkg/proxy/iptables/proxier.go

```diff
@@ -1442,19 +1428,6 @@ func (proxier *Proxier) syncProxyRules() {
 		}
 	}
 
-	// Drop the packets in INVALID state, which would potentially cause
-	// unexpected connection reset if nf_conntrack_tcp_be_liberal is not set.
-	// Ref: https://github.com/kubernetes/kubernetes/issues/74839
-	// Ref: https://github.com/kubernetes/kubernetes/issues/117924
-	if !proxier.conntrackTCPLiberal {
-		proxier.filterRules.Write(
-			"-A", string(kubeForwardChain),
-			"-m", "conntrack",
-			"--ctstate", "INVALID",
-			"-j", "DROP",
```

Comment:
Though while you're thinking about this... one of the other things in the nftables TODO was to add some more metrics to the iptables proxy to notice when users depend on iptables features/quirks that will go away with nftables.

The idea was that we could change `utiliptables.Interface.SaveInto()` to pass `--counters` to `iptables-save`, and then when we do a `SaveInto` down below, we can do some extra parsing to see how many packets this `--ctstate INVALID -j DROP` rule has acted on, and update a metric with that. And then, eventually, we'll have something that notices when the metric reaches a certain threshold, and alerts the user that they should consider `--conntrack-tcp-be-liberal` (and in particular, if they want to switch to nftables, they'll _need_ to use that).

Now, one problem with that idea is that we don't actually currently do `iptables-save` on the `filter` table, but that's not a lot of work to add...

(This is just an idea; it's not _necessary_ for this PR; it could still happen later even if you didn't do it now, and it might turn out to be more complicated than I'm imagining.)

Reply:
Will pick this up after completing this 🙂